### PR TITLE
XCSv2 deployment and updates

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

This PR prepares XCSv2 for instantiation on mainnet. Added a script (deploy_xcs.py) that fetches the data from the chain registry, gets it in the proper format and instantiates the registry, swaprouter, and xcs. 

As a drive-by, added the possibility of specifying the route to be used when making a swap (since it's been requested by a few teams)

## Testing and Verifying

All existing tests should pass (there are some issues with the rust tests atm)

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: cosmwasm/contracts/swaprouter/src/helpers.rs [hunk 1]
```diff
@@ -81,9 +81,15 @@ pub fn generate_swap_msg(
...
 ) -> Result<MsgSwapExactAmountIn, ContractError> {
-    // get trade route
-    let route = ROUTING_TABLE.load(deps.storage, (&input_token.denom, &min_output_token.denom))?;
+    let route = match route {
+        Some(route) => route,
+        None => {
+            // get trade route
+            ROUTING_TABLE.load(deps.storage, (&input_token.denom, &min_output_token.denom))?
+        }
+    };
```
Comment (ValarDragon) [inline cosmwasm/contracts/swaprouter/src/helpers.rs:92]: LGTM

**Review by ValarDragon — APPROVED** [review]
Agreed w/ go tests being where all the work is. LGTM!

**Review by PaddyMc — APPROVED** [review]
LGTM, tested with the updates to `osmosis-test-tube` and it works great :zap:

**Comment by anilhelvaci** [comment]
Hey @nicolaslara - I'm interested in building on top XCSv2 contracts. I was looking into the repository to understand how can I setup a local osmosis with a functional XCSv2. I thought CI could be a good place to start. I've noticed `e2e` tests do not cover XCSv2 and haven't been able detect `tests/ibc-hooks/ibc_middleware_test.go` being run anywhere in the CI. 

I thought `deploy_xcs.py` can be a guide however I'm not sure `tests/ibc-hooks` directory is actively maintained. Could you clarify how I should interpret this?

**Pull request merged**